autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/SchedulePreview.tsx

```diff
@@ -1,31 +1,112 @@
 import * as React from 'react';
-import { useSelector } from 'react-redux';
+import * as Cookies from 'js-cookie';
+import { useDispatch, useSelector } from 'react-redux';
 import { List } from '@material-ui/core';
 import GenericCard from '../../GenericCard/GenericCard';
 import { RootState } from '../../../redux/reducer';
 import * as styles from './SchedulePreview.css';
 import ScheduleListItem from './ScheduleListItem/ScheduleListItem';
 import Schedule from '../../../types/Schedule';
+import { setSchedules } from '../../../redux/actions/schedules';
+import createThrottleFunction from '../../../utils/createThrottleFunction';
+import { parseAllMeetings } from '../../../redux/actions/courseCards';
+import SmallFastProgress from '../../SmallFastProgress';
+
+const throttle = createThrottleFunction();
 
 export const noSchedulesText = "Click Generate Schedules to find schedules with the courses you've added.";
 
-const SchedulePreview: React.FC = () => {
+interface SchedulePreviewProps {
+  // Is a prop so we can set the throttle time to a really low number when testing
+  throttleTime?: number;
+  // Option to to hide the loading indicator, exclusively used for tests
+  hideLoadingIndicator?: boolean;
+}
+
+const SchedulePreview: React.FC<SchedulePreviewProps> = ({
+  throttleTime = 10000, hideLoadingIndicator = false,
+}) => {
+  const dispatch = useDispatch();
   const schedules = useSelector<RootState, Schedule[]>((state) => state.schedules);
+  const term = useSelector<RootState, string>((state) => state.term);
+  const [isLoadingSchedules, setIsLoadingSchedules] = React.useState(!hideLoadingIndicator);
 
   const scheduleListItems = schedules.length === 0
     ? <p className={styles.noSchedules}>{noSchedulesText}</p>
     : schedules.map((schedule, idx) => <ScheduleListItem index={idx} key={schedule.name} />);
 
+  // Whenever the term changes, fetch and load saved schedules
+  React.useEffect(() => {
+    if (term) {
+      fetch(`sessions/get_saved_schedules?term=${term}`).then(
+        (res) => res.json(),
+      ).then((arr: any[]): Schedule[] => arr.map((val) => ({
+        name: val.name,
+        meetings: parseAllMeetings(val.sections),
+        saved: true,
+      }))).then((savedSchedules: Schedule[]) => {
+        dispatch(setSchedules(savedSchedules));
+        setIsLoadingSchedules(false);
+      });
+    }
+
+    // on unmount, clear schedules
+    return (): void => {
+      dispatch(setSchedules([]));
+    };
+  }, [term, dispatch]);
+
+  // Call throttle to serialize and save the schedules anytime we make a change to the schedules
+  React.useEffect(() => {
+    if (!term) return;
+
+    // Serialize schedules and make API call
+    const saveSchedules = (): void => {
+      const serializedSchedules = schedules.filter(
```

Comment:
I'd still create an interface for this since it's being used for the tests and an interface would prevent headaches if we refactor this or add more properties to schedules in the future for whatever reason, it also makes the expected format more explicit